Sources/ZcashLightClientKit/Model/ScanRange.swift

```
@@ -18,7 +18,19 @@ struct ScanRange {
         case verify = 60
         
         init(_ value: UInt8) {
-            self = Priority(rawValue: value) ?? .ignored
+            // the given priority matches the values
+            if let priority = Priority(rawValue: value) {
+                self = priority
+            } else {
+                // no match found, getting the one above the value
+                let valueNeeded = (value / 10) * 10 + 10
+                
+                if let priority = Priority(rawValue: valueNeeded) {
+                    self = priority
+                } else {
+                    fatalError("The value \(value) is out of the range of priorities.")
+                }
+            }
```

I don't think that coercing here is the right solution; this is an enumeration, so failing to recognize the exact value _should_ just be a fatal error. So the previous implementation was correct, but instead of defaulting to `.ignored` it should raise the error.

---

Only the matching priorities are allowed, otherwise fatalError